src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Telemetry/TelemetryReporter.cs

```diff
@@ -115,13 +128,13 @@ public void ReportFault(Exception exception, string? message, params object?[] @
                 // We (potentially) have multiple exceptions; let's just report each of them
                 foreach (var innerException in aggregateException.Flatten().InnerExceptions)
                 {
-                    ReportFault(innerException, message, @params);
+                    ReportFault(innerException, message, depth + 1, @params);
```

Comment:
Does a call to ReportFault ever exist with a depth more than 1, that originates from this call? I would expect the `Flatten()` call above to remove any possibility for further recursion being necessary, as it completely flattens without recursion: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/AggregateException.cs,300

Reply:
- ReportFault call HandleException
- HandleException may call ReportRemoteInvocationException
- ReportRemoteInvocationException may call ReportFault again (for its innerException)

Reply:
Okay you're specifically asking about line 131. I'm not suggesting this line causes the SOE. Since I added  depth to the argument I updated this line.

Reply:
Ahh thank you, that's what I was missing. This could still overflow if an AggregateException contained a RemoteInvocationException, that contained more things.

I'd be tempted to solve this by just writing our own Flatten method that flattened out any of the exceptions we look inside of (AggregateException, RemoteInvocationException and OperationCanceledException). This could avoid recursion entirely.